cartridges/int_algolia/cartridge/scripts/algolia/model/algoliaLocalizedProduct.js

```diff
@@ -30,6 +30,28 @@ try {
 
 const ALGOLIA_IN_STOCK_THRESHOLD = algoliaData.getPreference('InStockThreshold');
 const INDEX_OUT_OF_STOCK = algoliaData.getPreference('IndexOutOfStock');
+const ATTRIBUTE_LIST = algoliaData.getSetOfArray('AdditionalAttributes');
+const stores = [];
+var SystemObjectMgr;
+var ProductInventoryMgr;
+
+if (ATTRIBUTE_LIST.indexOf('storeAvailability') !== -1) {
```

Comment:
Can this be moved outside this model? As you saw previously, accessing preferences and doing global things makes it harder to test.
We can imagine the jobs building the list of `Store` with an inventory list and passing it to the model:

```js
var storesList = [];
if (attributesToSend.indexOf('storeAvailability') !== -1) {
  var storeIt = SystemObjectMgr.getAllSystemObjects('Store');
  while (storeIt.hasNext()) {
    var store = storeIt.next();
    if (store.inventoryList) {
      storesList.push(store);
    }
  }
}
// ...
new AlgoliaLocalizedProduct({ storesList: storeList, ...
```

Reply:
Sure, I agree with you regarding it causing tests to be difficult, but we are calling AlgoliaLocalizedProduct on different files and places. In this case, we will write duplicated code, and I think this approach is cleaner.

Reply:
Fair enough